packages/core/src/engine/Autodiff.ts

```diff
@@ -939,16 +1045,16 @@ export const genCode = ({
   }
   const fields = [
     // somehow this actually works! if the gradient array is not contiguous, any
-    // gaps will just be filled in by commas when we call join, and that is
-    // valid JavaScript syntax for array literals with gaps, so everything works
-    // out perfectly
+    // holes will just be filled in by commas when we call join, and that is
+    // valid JavaScript syntax for array literals with holes, so everything
+    // works out perfectly
     `gradient: [${gradient.join(", ")}]`,
     `primary: ${primary}`,
     `secondary: [${secondary.join(", ")}]`,
   ];
   stmts.push(`return { ${fields.join(", ")} };`);
-  const f = new Function("inputs", stmts.join("\n"));
-  return (inputs) => f(inputs);
+  const f = new Function("polyRoots", "inputs", stmts.join("\n"));
```

Comment:
Interesting that this is how it works: passing function pointers to `Function`!

Reply:
Haha yes 😛 this seemed better than writing the body of the `polyRoots` function into our generated code every single time, because this way we get more typechecking.

Reply:
Yeah def better!